GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/Garage/GarageController.cs

```diff
@@ -90,4 +100,30 @@ private void SetCoinsBasedOnGameData()
     {
         coinsCount.text = GameDataController.GetGameData().CoinsOwned.ToString();
     }
+
+    public void OnRestorePurchaseButtonClicked()
+    {
+        restorePurchaseConfirmPanel.SetActive(true);
+    }
+
+    public void OnRestorePurchaseConfirmPanelButtonClicked(bool isConfirmed)
+    {
+        if (isConfirmed)
+        {
+            PurchaseController.RestorePurchase();
+        }
+
+        restorePurchaseConfirmPanel.SetActive(false);
+    }
+
+    public void OnRestorePurchaseSuccess()
```

Comment:
remind me do we provide listener for failure?

Reply:
This listener is written by me (developer). Yea, I should add a listener for failure too.